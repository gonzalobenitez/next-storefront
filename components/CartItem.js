import { FormattedNumber } from 'react-intl'
import Link from 'next/link'

export default ({ product, quantity, remove, setQuantity }) => {
  return (
  	<tr className="cart-item" >
  		<td>
			<Link href={{ pathname: '/product', query: { id: product.id } }}><a href={ `/product?id=${ product.id }` }><img src={ `/static/${ product.image }` } /></a></Link>
	  	</td>
  		<td>
		    <div>{ product.brand }</div>
		    <div>{ product.title }</div>
	    </td>
  		<td>
		    <input type="number" min="1" value={ quantity } onChange={ setQuantity } />
	    </td>
  		<td>
		    <span className="price"><FormattedNumber value={ product.price * quantity } style="currency" currency={ 'USD' } /></span>
	    </td>
  		<td>
		    <a onClick={ remove }>X</a>
	    </td>

		<style jsx>{`
			.cart-item  {
				text-align: left;
			}
			
	        .cart-item td:last-child {
	          text-align: right;
	        }

			.cart-item img {
				width: 10em;
    			height: 6em;
			}
		`}</style>
    </tr>
  )
}