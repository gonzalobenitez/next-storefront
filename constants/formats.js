export default {
  number: {
    USD: {
      style: 'currency',
      currency: 'USD'
    }
  }
}
