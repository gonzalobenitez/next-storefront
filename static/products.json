[
  {
    "title": "Blue Stripe Stoneware Plate",
    "brand": "Kiriko",
    "price": 40,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
    "image": "blue-stripe-stoneware-plate.jpg",
    "id": 12
  },
  {
    "title": "Hand Painted Blue Flat Dish",
    "brand": "Kiriko",
    "price": 28,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget arcu. Curabitur ac pharetra nisl, sit amet mattis dolor.",
    "image": "hand-painted-blue-flat-dish.jpg",
    "id": 22
  },
  {
    "title": "Heme",
    "brand": "Dust & Form",
    "price": 52,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget arcu. Curabitur ac pharetra nisl, sit amet mattis dolor.",
    "image": "heme.jpg",
    "id": 32
  },
  {
    "title": "Mashiko-Yaki Green Small Plate",
    "brand": "Kiriko",
    "price": 28,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
    "image": "mashiko-yaki-green-small-plate.jpg",
    "id": 42
  },
  {
    "title": "Mashiko-Yaki Indigo Small Plate",
    "brand": "Kiriko",
    "price": 28,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
    "image": "mashiko-yaki-indigo-small-plate.jpg",
    "id": 52
  },
  {
    "title": "Mashiko-Yaki Saucer",
    "brand": "Kiriko",
    "price": 18,
    "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam at purus pulvinar, placerat turpis ac, interdum metus. In eget massa sed enim hendrerit auctor a eget.",
    "image": "mashiko-yaki-saucer.jpg",
    "id": 62
  }
]
