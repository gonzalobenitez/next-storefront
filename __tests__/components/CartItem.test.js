import { shallow } from 'enzyme'
import { IntlProvider } from 'react-intl'
import renderer from 'react-test-renderer'
import CartItem from '../../components/CartItem.js'
import products from '../../data/products.json'
import formats from '../../constants/formats'
import Store from '../../stores/store'

describe('CartItem Test', () => {
  let store = null
  beforeEach(() => {
    store = new Store(products)
  })

  it('CartItem shows item', () => {
    const product = store.products[0]
    const component = shallow(<CartItem
      product={ product }
      quantity={ 1 }
      remove={() => store.removeFromCart(product.id, 1) }
      setQuantity={ (event) => store.setCartQuantity(product.id, event.target.value) }
    />)
    expect(component.find('input').props().value).toEqual(1)
  })

  it('CartItem shows item', () => {
    const product = store.products[0]
    const component = renderer.create(<IntlProvider locale="en" formats={ formats } defaultFormats={ formats } ><div><CartItem
      product={ product }
      quantity={ 1 }
      remove={() => store.removeFromCart(product.id, 1) }
      setQuantity={ (event) => store.setCartQuantity(product.id, event.target.value) }
    /></div></IntlProvider>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
