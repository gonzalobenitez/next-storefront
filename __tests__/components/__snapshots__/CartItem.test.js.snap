// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Test shows item in Snapshot Testing 1`] = `
<tr
  className="jsx-1118603124 cart-item"
>
  <td
    className="jsx-1118603124"
  >
    <a
      className="jsx-1118603124"
      href="/product?id=12"
      onClick={[Function]}
    >
      <img
        className="jsx-1118603124"
        src="/static/blue-stripe-stoneware-plate.jpg"
      />
    </a>
  </td>
  <td
    className="jsx-1118603124"
  >
    <div
      className="jsx-1118603124"
    >
      Kiriko
    </div>
    <div
      className="jsx-1118603124"
    >
      Blue Stripe Stoneware Plate
    </div>
  </td>
  <td
    className="jsx-1118603124"
  >
    <input
      className="jsx-1118603124"
      min="1"
      onChange={[Function]}
      type="number"
      value={1}
    />
  </td>
  <td
    className="jsx-1118603124"
  >
    <span
      className="jsx-1118603124 price"
    >
      <span>
        $40.00
      </span>
    </span>
  </td>
  <td
    className="jsx-1118603124"
  >
    <a
      className="jsx-1118603124"
      onClick={[Function]}
    >
      X
    </a>
  </td>
</tr>
`;
